cxx-squid/src/main/java/org/sonar/cxx/visitors/CxxParseErrorLoggerVisitor.java

```diff
@@ -36,28 +36,31 @@
   extends SquidAstVisitor<GRAMMAR> implements AstVisitor {
 
   private static final String SYNTAX_ERROR_MSG
-    = "Syntax error in a file detected. Syntax errors could cause invalid software metric values."
+    = "Source code parser: {} syntax error(s) detected. Syntax errors could cause invalid software metric values."
     + " Root cause are typically missing includes, missing macros or compiler specific extensions."
     + " Turn debug info on to get more details.";
   private static final Logger LOG = Loggers.get(CxxParseErrorLoggerVisitor.class);
   private final SquidAstVisitorContext<?> context;
-  private static boolean firstError = true;
+  private static int errors = 0;
 
   public CxxParseErrorLoggerVisitor(SquidAstVisitorContext<?> context) {
     this.context = context;
   }
 
+  public static void finalReport() {
+    if (!LOG.isDebugEnabled() && (errors > 0)) {
```

Comment:
I think it would be more consistent to log the WARN message always: 
instead of `if (!LOG.isDebugEnabled() && (errors > 0)) {` just `if (errors != 0)` ?